strings-csharp/StringTruncationInCSharp/StringTruncationInCSharp/StringHelper.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+
+namespace StringTruncationInCSharp
+{
+    public class StringHelper
+    {
+        public string TruncateStringUsingSubstring(string originalString, int maxLength)
+        {
+            if (maxLength <= 0)
+            {
+                return string.Empty;
+            }
+            else if (maxLength >= originalString.Length)
+            {
+                return originalString;
+            }
```

Comment:
In lots of methods, we have this check. We need to refactor this and extract these checks to a separate method.
We can achieve this by introducing a new private method:
`private string TruncateString(string originalString, int maxLength, Func<string, int, string> specificTruncateMethod)`
Inside of this method, we will have checks and the call of the specificTruncateMethod.
Next, in all the methods that use these checks, we need to have a local function that corresponds to the Func delegate form the above. For instance TruncateStringUsingSubstring will be like this:
```
           private string TruncateString(string originalString, int maxLength, Func<string, int, string> specificTruncateMethod)
        {
            if (maxLength <= 0)
            {
                return string.Empty;
            }
            else if (maxLength >= originalString.Length)
            {
                return originalString;
            }

            return specificTruncateMethod(originalString, maxLength);
        }
        public string TruncateStringUsingSubstring(string originalString, int maxLength)
        {
            return TruncateString(originalString, maxLength, UsingSubstring);

            string UsingSubstring(string str, int len)
            {
                return str.Substring(0, len);
            }
        }
```